### PR TITLE
Add Student.java entity file and Student.json migration file

### LINKED ISSUE #31: Subfeature: Add Student database Entity and Migration file
Part of Epic #6 

A subfeature of #12 

Since the endpoints for Student depends on the Course database, while we wait for the PR to get Course merged into main, we can  first add: 

- Student.java (Entity file)
- Student.json (Migration file)

### PR DESCRIPTION
**Pull request by AjayLiu**
## Overview
In this PR I add an entity file for the Student database table. I also added a corresponding migration file Student.json. This change is currently not accessible via the frontend or swagger.

## Tests

- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues

Closes #31

### CODE REVIEW

File: src/main/resources/db/migration/changes/Student.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                  }
+                },
+                {
+                  "column": {
+                    "name": "COURSE_ID",
+                    "type": "BIGINT"
+                  }
+                }
+              ],
+              "tableName": "students"
```
Comment (r0ckwav3) [inline src/main/resources/db/migration/changes/Student.json:67]: is the table name "student" or "students"? you use "student" earlier

**Review by r0ckwav3 — CHANGES_REQUESTED** [review]

**Review by r0ckwav3 — APPROVED** [review]
Looks good!

File: src/main/resources/db/migration/changes/Student.json (hunk 1 above)
Comment (AjayLiu) [reply src/main/resources/db/migration/changes/Student.json]: Good catch!

File: src/main/java/edu/ucsb/cs156/happiercows/repositories/StudentRepository.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  Optional<Student> findById(Long id);
+
+  @Query(value = "SELECT stu FROM student stu WHERE stu.email = :email")
+  Iterable<Student> findByEmail(String email);
+
+  @Query(value = "SELECT stu FROM student stu WHERE stu.courseId = :courseId")
+  Iterable<Student> findByCourseId(Long courseId);
+
+  @Query(value = "SELECT stu FROM student stu WHERE stu.perm = :perm")
+  Optional<Student> findByPerm(String perm);
```
Comment (r0ckwav3) [inline src/main/java/edu/ucsb/cs156/happiercows/repositories/StudentRepository.java:23]: I think this should return an iterable

File: src/main/java/edu/ucsb/cs156/happiercows/repositories/StudentRepository.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+import edu.ucsb.cs156.happiercows.entities.Student;
+
+@Repository
+public interface StudentRepository extends CrudRepository<Student, Long> {
+  @Query(value = "SELECT stu FROM student stu WHERE stu.id = :id")
+  Optional<Student> findById(Long id);
```
Comment (r0ckwav3) [inline src/main/java/edu/ucsb/cs156/happiercows/repositories/StudentRepository.java:14]: I think CrudRepository creates this method by default (see team02)

**Review by r0ckwav3 — CHANGES_REQUESTED** [review]
A couple of comments on the repository code

**Review by r0ckwav3 — APPROVED** [review]
LGTM!

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**